### PR TITLE
Add udp echo server example

### PR DESCRIPTION
**Pull request by apatrushev**

### CODE REVIEW

File: udp-echo/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+[package]
+name = "udp-echo"
+version = "0.1.0"
+authors = ["Anton Patrushev <apatrushev@gmail.com>"]
+edition = "2018"
+
+[dependencies]
+actix = "0.7"
+tokio = "*"
```
Comment (DoumanAsh) [inline udp-echo/Cargo.toml]: use `0.x` format for all deps, otherwise in future they'll break

File: udp-echo/src/main.rs [hunk 2]
```diff
@@ -0,0 +1,47 @@
+extern crate actix;
```
Comment (DoumanAsh) [inline udp-echo/src/main.rs]: with 2018 edition mode, you're not required to `extern crate` your deps

File: udp-echo/src/main.rs [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+impl Actor for UdpActor {
+    type Context = Context<Self>;
+}
+
+#[derive(Message)]
+struct UdpPacket(BytesMut, SocketAddr);
+impl StreamHandler<UdpPacket, std::io::Error> for UdpActor {
+    fn handle(&mut self, msg: UdpPacket, _: &mut Context<Self>) {
+        println!("Received: ({:?}, {:?})", msg.0, msg.1);
+        (&mut self.sink).send((msg.0.into(), msg.1)).wait().unwrap();
```
Comment (DoumanAsh) [inline udp-echo/src/main.rs]: If I'm not mistaken `Stream::send` should consume self? Does it compile?

Comment (apatrushev) [reply udp-echo/src/main.rs]: It is `Sink::send` not a `Stream::send`.
https://docs.rs/futures/0.1.25/futures/sink/trait.Sink.html#method.send

Comment (DoumanAsh) [reply udp-echo/src/main.rs]: Yeah ok, it doesn't change the fact that it takes `self` rather than `&mut self`

Comment (apatrushev) [reply udp-echo/src/main.rs]: Interesting. But this code works somehow. :(

Comment (apatrushev) [reply udp-echo/src/main.rs]: Ok, the travis checks passed showing that this code compiles. Should I fix it if it works? I do not like the situation when I do not understand why it works and will be happy to find explanation, but I definitely need a help.

Comment (DoumanAsh) [reply udp-echo/src/main.rs]: If it compiles then everything is ok I guess. which is surprising but fine too

**Comment by svenstaro** [comment]
Can you rebase this?

**Comment by apatrushev** [comment]
> Can you rebase this?

Do you plan to merge it? I will do rebase in that case.

**Comment by svenstaro** [comment]
I think it's a useful example and I'd like to merge it, yes.

**Comment by apatrushev** [comment]
> I think it's a useful example and I'd like to merge it, yes.

Done.

**Comment by svenstaro** [comment]
Thanks!

**Pull request merged**